### PR TITLE
[Python] Fix `relation.query()` not accepting non-select statements

### LINKED ISSUE #5380: Python API no longer allows "COPY" statements in v0.6
### What happens?

Prior to version 0.6, `DuckDBPyRelation.query` supported the `COPY` statement.  I used this to perform some ETL work on parquet files in a data pipeline (parquet input, transform, parquet and csv output).  After upgrading to `0.6`, this now produces an `InvalidInputException`.

### To Reproduce

My own usecase is far more complicated, but here is the minimum needed to reproduce.

```python
import duckdb
from pathlib import Path

local_path = Path("/tmp/test.parq")

rel = duckdb.from_parquet(str(local_path))

# Select subset of columns and write output to new parquet file.
rel.project("col_a, col_b, col_c").query(
    "tmp_table", 
    """
    COPY (SELECT * FROM tmp_table)
    TO '/tmp/out.parq' (FORMAT 'PARQUET', CODEC 'SNAPPY')
    """
)



InvalidInputException                     Traceback (most recent call last)
Cell In [19], line 1
----> 1 rel.project("first_name, last_name").query(
      2     "tmp_table",
      3     """
      4     COPY (SELECT * FROM tmp_table)
      5     TO '/tmp/out.parq' (FORMAT 'PARQUET', CODEC 'SNAPPY')
      6     """
      7 )

InvalidInputException: Invalid Input Error: 'DuckDBPyRelation.query' does not accept statements of type COPY



```

### OS:

MacOS and Linux (arm)

### DuckDB Version:

0.6.0

### DuckDB Client:

Python

### Full Name:

Erik Wickstrom

### Affiliation:

LiftCertain

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #5380 

Instead of throwing an error, we now still execute the query (because of the side-effects it could have), and return None as the DuckDBPyRelation result of the function

### CODE REVIEW

File: tools/pythonpkg/src/pyrelation.cpp [hunk 1]
```diff
@@ -659,8 +659,14 @@ unique_ptr<DuckDBPyRelation> DuckDBPyRelation::Query(const string &view_name, co
...
 		return Query(view_name, query);
 	}
-	throw InvalidInputException("'DuckDBPyRelation.query' does not accept statements of type %s",
-	                            StatementTypeToString(statement.type));
+	{
+		py::gil_scoped_release release;
+		auto query_result = rel->context.GetContext()->Query(move(parser.statements[0]), false);
+		// Execute it anyways, for creation/altering statements
+		// We only care that it succeeds, we can't store the result
+		D_ASSERT(query_result);
```
Comment (Mytherin) [inline tools/pythonpkg/src/pyrelation.cpp:667]: Should we not check the result here and throw an exception if it does not? Could we add a test with a failing query (e.g. with a table that does not exist)?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM. One comment:

File: tools/pythonpkg/src/pyrelation.cpp (hunk 1 above)
Comment (Tishj) [reply tools/pythonpkg/src/pyrelation.cpp:667]: Ah fair enough, had not thought about it, other than the assert
I blame monday
Will fix it now :)

**Comment by Tishj** [comment]
I'm guessing this should also go straight to master, seeing as it's a bug fix?

**Comment by Mytherin** [comment]
Yes, good point

**Pull request closed without merging**